### PR TITLE
[tests-only][full-ci] Test scenarios for search/REPORT response for personal shares

### PR DESCRIPTION
**Pull request by roshanlc**
## Description
This PR adds tests for checking REPORT response of searched file/folder for personal shares.

## Related Issue

- Related to https://github.com/owncloud/ocis/issues/4489

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiContract/personalShareReport.feature [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      | Brian    |
+
+
+  Scenario Outline: Check the REPORT response of the found folder
+    Given using <dav_version> DAV path
+    And user "Alice" has created folder "/folderMain"
+    And user "Alice" has created folder "/folderMain/SubFolder1"
+    And user "Alice" has created folder "/folderMain/SubFolder1/subFOLDER2"
+    And user "Alice" has shared entry "/folderMain" with user "Brian" with permissions 17
+    And user "Brian" has accepted share "/folderMain" offered by user "Alice"
```
Comment (grgprarup) [inline tests/acceptance/features/apiContract/personalShareReport.feature:18]: These steps can be put in the background.

File: tests/acceptance/features/apiContract/personalShareReport.feature [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      | old         |
+      | new         |
+
+
+  Scenario Outline: Check the REPORT response of the found file
+    Given using <dav_version> DAV path
+    And user "Alice" has created folder "/folderMain"
+    And user "Alice" has created folder "/folderMain/SubFolder1"
+    And user "Alice" has created folder "/folderMain/SubFolder1/subFOLDER2"
+    And user "Alice" uploads file with content "Not all those who wander are lost." to "/folderMain/SubFolder1/subFOLDER2/frodo.txt" using the WebDAV API
```
Comment (grgprarup) [inline tests/acceptance/features/apiContract/personalShareReport.feature:40]:
```suggestion
    And user "Alice" has uploaded file with content "Not all those who wander are lost." to "/folderMain/SubFolder1/subFOLDER2/frodo.txt"
```

File: tests/acceptance/features/apiContract/personalShareReport.feature [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+    And user "Alice" has created folder "/folderMain"
+    And user "Alice" has created folder "/folderMain/SubFolder1"
+    And user "Alice" has created folder "/folderMain/SubFolder1/subFOLDER2"
+    And user "Alice" has shared entry "/folderMain" with user "Brian" with permissions 17
```
Comment (grgprarup) [inline tests/acceptance/features/apiContract/personalShareReport.feature:13]:
```suggestion
    And user "Alice" has shared entry "/folderMain" with user "Brian" with permissions "17"
```

**Comment by SwikritiT** [comment]
@roshanlc I think these can be added in this file? https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiContract/report.feature

**Comment by roshanlc** [comment]
> @roshanlc I think these can be added in this file? https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiContract/report.feature

`../apiContract/report.feature` contains tests for search/REPORT response for a dedicated space while this PR adds tests for search/REPORT response for searching in personal shares.  I think it is better to have separate feature file for search/REPORT response of personal shares as the background steps are quite different.

**Comment by SwikritiT** [comment]
> > @roshanlc I think these can be added in this file? https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/apiContract/report.feature
> 
> `../apiContract/report.feature` contains tests for search/REPORT response for a dedicated space while this PR adds tests for search/REPORT response for searching in personal shares. I think it is better to have separate feature file for search/REPORT response of personal shares as the background steps are quite different.

then rename the other file to spacesReport or something similar so it's specific

**Review by grgprarup — COMMENTED** [review]
LGTM :+1:

**Review by SwikritiT — APPROVED** [review]
LGTM

File: tests/acceptance/features/apiContract/personalShareReport.feature [hunk 4]
```diff
@@ -0,0 +1,52 @@
+@api
+Feature: Test REPORT response for personal shares
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/personalShareReport.feature:2]:
```suggestion
Feature: REPORT request Shares space
```

File: tests/acceptance/features/apiContract/personalShareReport.feature [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  Check that the REPORT response contains all relevant details for personal shares
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/personalShareReport.feature:3]:
```suggestion
  Check that the REPORT response contains all relevant details for Shares
```

File: tests/acceptance/features/apiContract/personalShareReport.feature (hunk 4 above)
Comment (saw-jan) [inline tests/acceptance/features/apiContract/personalShareReport.feature:2]: file can be renamed to `personalShareReport.feature` -> `sharesReport.feature`

**Review by phil-davis — APPROVED** [review]

**Comment by SwikritiT** [comment]
@roshanlc please make a PR to master with these changes

**Pull request merged**